### PR TITLE
fix install kubectl

### PR DESCRIPTION
**Pull request by vhvb1989**
fix: https://github.com/Azure/azure-dev/issues/3471

using instructions from: https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/#install-using-native-package-management

### CODE REVIEW

**Review by danieljurek — APPROVED** [review]
I approve of this provided we file an issue to update kubectl version so this doesn't get stale. 

We could also consider putting in a call to `$(curl -L -s https://dl.k8s.io/release/stable.txt)` ... but not sure if the package manager URLs map nicely to specific versions as opposed to `<major>.<minor>`.

File: cli/Dockerfile [hunk 1]
```diff
@@ -40,8 +40,8 @@ RUN export DEBIAN_FRONTEND=noninteractive \
      # kubectl CLI
      && apt-get update \
      && mkdir -p /etc/apt/keyrings \
-     && curl -fsSLo /etc/apt/keyrings/kubernetes-archive-keyring.gpg https://dl.k8s.io/apt/doc/apt-key.gpg \
-     && echo "deb [signed-by=/etc/apt/keyrings/kubernetes-archive-keyring.gpg] https://apt.kubernetes.io/ kubernetes-xenial main" | tee /etc/apt/sources.list.d/kubernetes.list \
+     && curl -fsSL https://pkgs.k8s.io/core:/stable:/v1.29/deb/Release.key | gpg --dearmor -o /etc/apt/keyrings/kubernetes-apt-keyring.gpg \
+     && echo 'deb [signed-by=/etc/apt/keyrings/kubernetes-apt-keyring.gpg] https://pkgs.k8s.io/core:/stable:/v1.29/deb/ /' | tee /etc/apt/sources.list.d/kubernetes.list \
```
Comment (wbreza) [inline cli/Dockerfile:44]: I'm hesitant to using a pinned version here and would prefer we use one of the latest versions mentioned @ https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/#install-using-native-package-management

Comment (wbreza) [reply cli/Dockerfile:44]: Please see PR #3474 for pin-less strategy.

**Comment by wbreza** [comment]
Closed in favor of #3474

**Pull request closed without merging**